### PR TITLE
Show character count on new w/o description

### PR DESCRIPTION
**Pull request by cdccollins**
https://trello.com/c/sFsNTtm2/197-as-an-rcc-agent-i-need-to-see-how-many-characters-are-in-the-description-so-that-i-know-what-an-operative-and-a-scheduler-cannot

### CODE REVIEW

File: app/views/repair_requests/new.html.erb [hunk 1]
```diff
@@ -86,3 +88,13 @@
     </div>
   </div>
 </div>
+
+<script>
```
Comment (pudiva) [inline app/views/repair_requests/new.html.erb]: Can you please move this to a .js file?

Comment (pudiva) [reply app/views/repair_requests/new.html.erb]: Also, don't we jave jQuery available? It used to come with Rails

File: spec/features/repair_request_spec.rb [hunk 2]
```diff
@@ -278,7 +278,7 @@ def stub_property_temp_annex
 
       visit new_property_repair_request_path('00000666', status: "1", guid: '123456789')
 
-      fill_in "Problem description", with: "It's broken"
+      fill_in "description-text", with: "It's broken"
```
Comment (pudiva) [inline spec/features/repair_request_spec.rb]: Why is this necessary?

File: app/views/repair_requests/new.html.erb [hunk 3]
```diff
@@ -56,15 +56,17 @@
         ) %>
       <% end %>
 
-      <%= form.form_group :description do %>
+      <%= form.form_group :description  do %>
```
Comment (pudiva) [inline app/views/repair_requests/new.html.erb]: Please, can you amend this commit and remove the whitespace change?

File: app/assets/stylesheets/hackney_new_repair_request.scss [hunk 4]
```diff
@@ -0,0 +1,3 @@
+.hackney-repair-request-description-hint {
+  margin-top: -20px;
```
Comment (pudiva) [inline app/assets/stylesheets/hackney_new_repair_request.scss]: Do we really need that margin now? And can't it be done with `govuk-` margin classes?

**Review by pudiva — CHANGES_REQUESTED** [review]

File: spec/features/repair_request_spec.rb (hunk 2 above)
Comment (cdccollins) [reply spec/features/repair_request_spec.rb]: Otherwise I get this 

`
Capybara::ElementNotFound:
       Unable to find field "Problem description" that is not disabled
`

File: app.json [hunk 5]
```diff
@@ -38,6 +38,8 @@
     },
     "X_API_KEY": {
       "required": true
+    }, "DRS_URL": {
+      "required": true
```
Comment (pudiva) [inline app.json:43]: That is interesting! I didn't know about this file!

File: app/assets/stylesheets/hackney_new_repair_request.scss (hunk 4 above)
Comment (cdccollins) [reply app/assets/stylesheets/hackney_new_repair_request.scss]: I'll use the govuk- margin class, always forget about that! I think we do need it, if we're following the general gds layout for this kind of thing

File: app.json (hunk 5 above)
Comment (cdccollins) [reply app.json:43]: Yeah, Ben W showed me this sneaky trick :)

File: app/views/repair_requests/new.html.erb [hunk 6]
```diff
@@ -86,3 +88,13 @@
     </div>
   </div>
 </div>
+
+<script>
+  var descriptionText = document.getElementById("description-text");
+  var counter = document.getElementById("counter");
+  var maxLength = counter.dataset.maximumLength;
+
+  descriptionText.addEventListener("keyup", function() {
```
Comment (pudiva) [inline app/views/repair_requests/new.html.erb]: When you move this to a .js file, you'll also need to change this into a `onkeyup=` attribute on the textarea.

I'm not sure if this is a good practice, but it's being used everywhere in the code, so we should stick with it unless we find a good reason not to.

Comment (cdccollins) [reply app/views/repair_requests/new.html.erb]: Yep, changing it now

File: app/assets/stylesheets/hackney_new_repair_request.scss (hunk 4 above)
Comment (pudiva) [reply app/assets/stylesheets/hackney_new_repair_request.scss]: Ohh so you copied this from the latest version?

Comment (cdccollins) [reply app/assets/stylesheets/hackney_new_repair_request.scss]: Screenshot 2019-06-19 at 10 56 27
I'm following this design from GDS

File: spec/features/repair_request_spec.rb (hunk 2 above)
Comment (pudiva) [reply spec/features/repair_request_spec.rb]: Strange. Looks like something is broken.

Comment (pudiva) [reply spec/features/repair_request_spec.rb]: Yes, the problem is that you specified an id to the textarea, when Rails' form helpers set it to tie it to the label. You must either use Rails-generated id or pass your change to other helpers that are related to it.

Comment (pudiva) [reply spec/features/repair_request_spec.rb]: Please, do that and revert the change to the spec, as users will interact with something labeled description, not with something have specific id.

Comment (pudiva) [reply spec/features/repair_request_spec.rb]: @cdccollins :point_up:

**Review by pudiva — APPROVED** [review]
:rainbow:
